test/swar/BasicOperations.cpp

```diff
@@ -6,15 +6,198 @@
+
+static_assert(0x80 == clearLSBits<4, u8>(0xF8));
+static_assert(0xC0 == clearLSBits<4, u8>(0xF4));
+static_assert(0xE0 == clearLSBits<4, u8>(0xF2));
+static_assert(0xF0 == clearLSBits<4, u8>(0xF1));
+
+static_assert(0xF0 == clearLSBits<4, u8>(0xFF));
+static_assert(0xC0 == clearLSBits<4, u8>(0xFC));
+static_assert(0xE0 == clearLSBits<4, u8>(0xFA));
+
+static_assert(0x80 == isolateLSBits<2, u8>(0x80));
+static_assert(0xC0 == isolateLSBits<2, u8>(0xC0));
+static_assert(0x60 == isolateLSBits<2, u8>(0xE0));
+static_assert(0x30 == isolateLSBits<2, u8>(0xF0));
+static_assert(0x18 == isolateLSBits<2, u8>(0xF8));
+static_assert(0x03 == isolateLSBits<2, u8>(0xFB));
+static_assert(0x0C == isolateLSBits<2, u8>(0xFC));
+static_assert(0x03 == isolateLSBits<2, u8>(0xFF));
+
+static_assert(0x0606'0606 == u32(broadcast<8>(SWAR<8, u32>(0x0000'0006))));
+static_assert(0x0808'0808 == u32(broadcast<8>(SWAR<8, u32>(0x0000'0008))));
+static_assert(0x0B0B'0B0B == u32(broadcast<8>(SWAR<8, u32>(0x0000'000B))));
+static_assert(0x0E0E'0E0E == u32(broadcast<8>(SWAR<8, u32>(0x0000'000E))));
+static_assert(0x6B6B'6B6B == u32(broadcast<8>(SWAR<8, u32>(0x0000'006B))));
+static_assert(0x0808'0808'0808'0808ull == u64(broadcast<8>(SWAR<8, u64>(0x0000'0000'0000'0008ull))));
+
+static_assert(2 == lsbIndex(1<<1));
+static_assert(4 == lsbIndex(1<<3));
+static_assert(6 == lsbIndex(1<<5));
+static_assert(9 == lsbIndex(1<<8));
+static_assert(18 == lsbIndex(1<<17));
+static_assert(31 == lsbIndex(1<<30));
+
+static_assert(0x80880008 == greaterEqual<3>(SWAR<4, uint32_t>(0x3245'1027)).value());
+static_assert(0x88888888 == greaterEqual<0>(SWAR<4, uint32_t>(0x0123'4567)).value());
+static_assert(0x88888888 == greaterEqual<0>(SWAR<4, uint32_t>(0x7654'3210)).value());
+static_assert(0x00000008 == greaterEqual<7>(SWAR<4, uint32_t>(0x0123'4567)).value());
+static_assert(0x80000000 == greaterEqual<7>(SWAR<4, uint32_t>(0x7654'3210)).value());
+
+
+// Unusual formatting for easy visual verification.
+#define GE_MSB_TEST(left, right, result) static_assert(result== greaterEqual_MSB_off<4, u32>(SWAR<4, u32>(left), SWAR<4, u32>(right)).value());
+
+GE_MSB_TEST(0x1000'0010,
+            0x0111'1101,
+            0x8000'0080)
+GE_MSB_TEST(0x4333'3343,
+            0x4444'4444,
+            0x8000'0080)
+
+GE_MSB_TEST(0x0123'4567,
+            0x0000'0000,
+            0x8888'8888); 
+GE_MSB_TEST(0x0123'4567,
+            0x0000'0000,
+            0x8888'8888)
+GE_MSB_TEST(0x0123'4567,
+            0x1111'1111,
+            0x0888'8888)
+GE_MSB_TEST(0x0123'4567,
+            0x2222'2222,
+            0x0088'8888)
+GE_MSB_TEST(0x0123'4567,
+            0x3333'3333,
+            0x0008'8888)
+GE_MSB_TEST(0x0123'4567,
+            0x4444'4444,
+            0x0000'8888)
+GE_MSB_TEST(0x0123'4567,
+            0x5555'5555,
+            0x0000'0888)
+GE_MSB_TEST(0x0123'4567,
+            0x6666'6666,
+            0x0000'0088)
+GE_MSB_TEST(0x0123'4567,
+            0x7777'7777,
+            0x0000'0008)
```

Comment:
Thanks for the attempt to be comprehensive, but alas, this is not!:
We ought to look for a "problem" of a "lane" that might "borrow" from the adjoining lane, or failing to compare against zero, or something like that.
Also, tests that are entirely redundant actually reduce clarity.  Please identify the cases you are testing, whether the tests are comprehensive, and do not introduce them redundantly

Reply:
Is fine, if you don't like redundancy, easy to cut down.  Very easy to generate, also.
These are all testing MSB off, so I'm not sure what edge case is interesting?  Lanes should be isolated, but shifted tests some.